src/anemoi/inference/inputs/ekd.py

```diff
@@ -150,15 +143,15 @@ def _create_input_state(
             name, valid_datetime = field.metadata("name"), field.metadata("valid_datetime")
             if name not in fields:
                 fields[name] = np.full(
-                    shape=(len(dates), self.checkpoint.number_of_grid_points),
+                    shape=(len(dates), n_points),
                     fill_value=np.nan,
                     dtype=dtype,
                 )
 
             date_idx = date_to_index[valid_datetime]
 
             try:
-                fields[name][date_idx] = mask.apply(field.to_numpy(dtype=dtype, flatten=flatten))
+                fields[name][date_idx] = field.to_numpy(dtype=dtype, flatten=flatten)[mask]
```

Comment:
In the case of no mask being supplied, this will apply a mask equal to the number of grid points (or a slice of None). Does this introduce an extra operation, or is numpy smart enough to treat that as a noop?

Before with the mask class and `apply()` we were sure that not having a mask would be a noop. We should be sure that that is still the case.

Reply:
If no mask is provided, mask is `slice(None)` which I would assume is interpreted as a noop by numpy.  But yeah, I am not 100% sure. This small test would suggest so:

```python
import numpy as np
x = np.arange(10)
y = x[range(10)]
print(np.shares_memory(x, y)) # False
print(np.all(x == y)) # True
y = x[slice(None)]
print(np.shares_memory(x, y)) # True
print(np.all(x == y)) # True
```